package/src/components/ChannelPreview/hooks/__tests__/useLatestMessagePreview.test.tsx

```diff
@@ -0,0 +1,317 @@
+  let chatClient: StreamChat<DefaultGenerics> | StreamChat<DefaultStreamChatGenerics>;
+
+  beforeEach(async () => {
+    chatClient = await getTestClientWithUser(clientUser);
+  });
+
+  const ChatProvider: FC<{ children: React.ReactNode }> = ({ children }) => (
+    <ChatContext.Provider
+      value={
+        {
+          auto_translation_enabled: true,
+          client: chatClient,
+        } as unknown as ChatContextValue
+      }
+    >
+      {children}
+    </ChatContext.Provider>
+  );
+
+  it('should return a channel latest message', async () => {
+    const { result } = renderHook(
+      () => useLatestMessagePreview(CHANNEL, FORCE_UPDATE, LATEST_MESSAGE),
+      { wrapper: ChatProvider },
+    );
+    await waitFor(() => {
+      expect(result.current).toBeTruthy();
+    });
+  });
+
+  it('should return a channel', async () => {
+    const { result } = renderHook(
+      () => useLatestMessagePreview(CHANNEL_WITH_MESSAGES, FORCE_UPDATE, LATEST_MESSAGE),
+      { wrapper: ChatProvider },
+    );
+    await waitFor(() => {
+      expect(result.current).toEqual({
+        created_at: 'L',
+        messageObject: {
+          created_at: new Date('2021-02-12T12:12:35.862Z'),
+          id: '',
+          message: {},
+          pinned_at: new Date('2021-02-12T12:12:35.862Z'),
+          status: 'received',
+          updated_at: new Date('2021-02-12T12:12:35.862Z'),
+        },
+        previews: [
+          { bold: false, text: '' },
+          { bold: false, text: 'Empty message...' },
+        ],
+        status: 0,
+      });
+    });
+  });
+
+  it('should return a nothing yet', async () => {
+    const { result } = renderHook(
+      () => useLatestMessagePreview(CHANNEL_WITH_NO_MESSAGES, FORCE_UPDATE, LATEST_MESSAGE),
+      { wrapper: ChatProvider },
+    );
+    await waitFor(() => {
+      expect(result.current).toEqual({
+        created_at: 'LT',
+        previews: [{ bold: false, text: 'Nothing yet...' }],
+        status: 0,
+      });
+    });
+  });
+
+  it('should return a deleted message', async () => {
+    const { result } = renderHook(
+      () => useLatestMessagePreview(CHANNEL_WITH_DELETED_MESSAGES, FORCE_UPDATE, LATEST_MESSAGE),
+      { wrapper: ChatProvider },
+    );
+    await waitFor(() => {
+      expect(result.current).toEqual({
+        created_at: 'LT',
+        messageObject: {
+          type: 'deleted',
+        },
+        previews: [{ bold: false, text: 'Message deleted' }],
+        status: 0,
+      });
+    });
+  });
+
+  it('should return message deleted', async () => {
+    const { result } = renderHook(
+      () => useLatestMessagePreview(CHANNEL_WITH_NO_MESSAGES, FORCE_UPDATE, LATEST_MESSAGE),
+      { wrapper: ChatProvider },
+    );
+    await waitFor(() => {
+      expect(result.current).toEqual({
+        created_at: 'LT',
+        previews: [{ bold: false, text: 'Nothing yet...' }],
+        status: 0,
+      });
+    });
+  });
+
+  it('should return latest text', async () => {
```

Comment:
The title of the test should be more specific, we're expecting the whole result is there.

But at the end of the day, we care about what is being rendered to the user.

Reply:
we say "should return latest text" if that's what you want to test, just expect `result.current.reviews` (or whatever is being rendered)